server/etcdserver/version/version.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package version
+
+import (
+	"github.com/coreos/go-semver/semver"
+	"go.etcd.io/bbolt"
+
+	"go.etcd.io/etcd/server/v3/mvcc/backend"
+	"go.etcd.io/etcd/server/v3/mvcc/buckets"
+)
+
+// ReadLocalVersion loads storage local etcd version from given backend transaction.
+func ReadLocalVersion(tx backend.ReadTx) string {
+	tx.Lock()
+	defer tx.Unlock()
+	return unsafeReadLocalVersion(tx)
+}
+
+// ReadLocalVersionFromSnapshot loads storage local etcd version from given bbolt transaction.
+func ReadLocalVersionFromSnapshot(tx *bbolt.Tx) string {
+	v := tx.Bucket(buckets.Meta.Name()).Get(buckets.MetaLocalVersionName)
+	version, err := semver.NewVersion(string(v))
+	if err != nil {
+		return ""
+	}
+	return version.String()
+}
+
+func unsafeReadLocalVersion(tx backend.ReadTx) string {
+	_, vs := tx.UnsafeRange(buckets.Meta, buckets.MetaLocalVersionName, nil, 1)
+	if len(vs) == 0 {
+		return ""
+	}
```

Comment:
Panic if >1 ?

Reply:
As only using `vs[0]` we don't need to check for more elements to avoid panic. Adding `len(vs) > 0` would be a basically an assert check checks `UnsafeRange` works correctly. It's not needed and reduced code readability.